### PR TITLE
Issue #12984: Add schedular to update github-actions

### PR DESCRIPTION
**Pull request by AayushSaini101**
Related:  https://github.com/checkstyle/checkstyle/issues/12984

### CODE REVIEW

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by AayushSaini101** [comment]
> Ok to merge

Please add hacktoberfest-accepted

File: .github/dependabot.yml
Comment (nrmancuso) [inline .github/dependabot.yml:14]: Please add commit message prefix as we do in maven config above.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: .github/dependabot.yml
Comment (AayushSaini101) [reply .github/dependabot.yml:14]: Sure @nrmancuso

Comment (AayushSaini101) [reply .github/dependabot.yml:14]: @nrmancuso Added commit message.

Comment (AayushSaini101) [reply .github/dependabot.yml:14]: Done

Comment (nrmancuso) [reply .github/dependabot.yml:14]: This will cause CI to fail, we do not allow this commit prefix. @romani @rnveach are we good with just using `dependency` prefix for actions dependencies too? I am.

Comment (rnveach) [reply .github/dependabot.yml:14]: I am fine with `dependency` being used for anything representing an external version upgrade.

Comment (nrmancuso) [reply .github/dependabot.yml:14]: Ok, @AayushSaini101 lets just reuse `dependency` prefix, after all, it is “dependabot”

Comment (AayushSaini101) [reply .github/dependabot.yml:14]: Done @nrmancuso

Comment (romani) [reply .github/dependabot.yml:14]: @AayushSaini101 , please do not resolve conversations, it should b done by maintainers

**Review by nrmancuso — APPROVED** [review]
Good to merge once CI is green, I reran some failed jobs.

**Comment by romani** [comment]
sonar can be ignored in this PR, but problem with CI will be blocker  at
https://github.com/checkstyle/checkstyle/pull/13844#issuecomment-1752046866
and similar PRs with java functional code changes.

**Comment by nrmancuso** [comment]
@AayushSaini101 why does it look like your commits are made by another user? I think this is causing sonar to fail to find environment variables

**Comment by AayushSaini101** [comment]
> @AayushSaini101 why does it look like your commits are made by another user? I think this is causing sonar to fail to find environment variables

Replied in private chat

**Comment by nrmancuso** [comment]
@AayushSaini101 as Romani mentioned above, this will become a problem in functional PRs.

**Comment by AayushSaini101** [comment]
> @AayushSaini101 as Romani mentioned above, this will become a problem in functional PRs.

I tried using a suitable username but still this is failing

**Comment by nrmancuso** [comment]
CircleCI is still running on the other account, see https://app.circleci.com/jobs/github/AayushSaini101/checkstyle/1365

We can ignore in this PR, but for future PRs let’s try to use main account and see if this helps.

**Comment by nrmancuso** [comment]
Let’s try in next PR, maybe we are affected by PR being originally sent from the other account.

**Comment by rnveach** [comment]
@AayushSaini101 Please make CI green.

**Comment by AayushSaini101** [comment]
@rnveach can I try to open new PR for this issue ? I have changed the account user that might causing fails

**Comment by romani** [comment]
Yes you can try another user, all we need is CI to pass.

**Comment by rnveach** [comment]
If you can't re-push to the original branch, then be welcome opening a new PR. Just ping us here on the new one.

**Comment by romani** [comment]
@rnveach , please approve PR, this specific PR we can merge without sonar, as it does not validate yml

File: .github/dependabot.yml
Comment (rnveach) [inline .github/dependabot.yml:17]: I would just make this the same as the other and add `open-pull-requests-limit: 10`

**Review by rnveach — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml
Comment (AayushSaini101) [reply .github/dependabot.yml:17]: Done

**Comment by rnveach** [comment]
@AayushSaini101 I would suggest updating your copy of main repo's master. You keep using an outdated version.

> The PR is based on a master that is 21 commit(s) old.
The max allowed is 10.

**Comment by AayushSaini101** [comment]
> @AayushSaini101 I would suggest updating your copy of main repo's master. You keep using an outdated version.
> 
> > The PR is based on a master that is 21 commit(s) old.
> > The max allowed is 10.

Okay thanks for the instruction.

**Comment by AayushSaini101** [comment]
> @AayushSaini101 I would suggest updating your copy of main repo's master. You keep using an outdated version.
> 
> > The PR is based on a master that is 21 commit(s) old.
> > The max allowed is 10.

Done rebase master and push

**Review by rnveach — APPROVED** [review]

**Pull request merged**
